### ISSUE TITLE
Failed in generating report data when test code throws exception

### ISSUE THREAD

**Issue by alex-vungle** [cenfun/monocart-reporter#25]
**Monocart version:** 1.6.20

I noticed if the test code throws unhandled exceptions directly:

```
......

test("foo", async () => {
    throw "Invalid error";
})

......
```

The monocart report will not able to generate report data with below error:

```
[MCR] generating report data ...
Error in reporter TypeError: err.split is not a function
    at /Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/visitor.js:648:31
    at Array.map (<anonymous>)
    at Visitor.errorsToSnippets (/Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/visitor.js:647:23)
    at /Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/visitor.js:617:36
    at forList (/Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/platform/share.js:116:32)
    at forList (/Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/platform/share.js:120:35)
    at forList (/Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/platform/share.js:120:35)
    at Object.forEach (/Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/platform/share.js:126:9)
    at Visitor.duplicatedErrorsHandler (/Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/visitor.js:597:14)
    at Visitor.start (/Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/visitor.js:52:14)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at generateData (/Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/generate-data.js:44:5)
    at Reporter.onEnd (/Users/ali/workspace/WebQA-Playwright/node_modules/monocart-reporter/lib/index.js:127:28)
    at Multiplexer.onExit (/Users/ali/workspace/WebQA-Playwright/node_modules/@playwright/test/lib/reporters/multiplexer.js:102:45)
    at Runner.runAllTests (/Users/ali/workspace/WebQA-Playwright/node_modules/@playwright/test/lib/runner/runner.js:75:5)
    at runTests (/Users/ali/workspace/WebQA-Playwright/node_modules/@playwright/test/lib/cli.js:135:138)
    at ji.<anonymous> (/Users/ali/workspace/WebQA-Playwright/node_modules/@playwright/test/lib/cli.js:48:7)
```

The Playwright HTML report is able to handle this scenario gracefully:

![image](https://github.com/cenfun/monocart-reporter/assets/42924996/43b0abdd-35c5-4995-ae76-b9b736224dd0)

**Comment by cenfun** [comment]
good catch, will be fixed next release.
currently, you could replace it with
```js
test('foo', () => {
    throw new Error('Invalid error');
});
```
It is considered good practice to only throw the Error object not string. 
and recommended to use `eslint` to improve the code
![image](https://github.com/cenfun/monocart-reporter/assets/334577/d0cd17df-b0da-45da-a196-0e3586ca352f)
see rule [no-throw-literal]https://eslint.org/docs/latest/rules/no-throw-literal

**Comment by cenfun** [comment]
fixed in monocart-reporter@1.6.22
